admin/services/data-access/pin-generation.data.service.js

```diff
@@ -11,12 +11,48 @@ const serviceToExport = {
    * @return {Promise<*>}
    */
   sqlFindEligiblePupilsBySchool: async (schoolId, isLiveCheck) => {
-    const view = isLiveCheck === true ? 'vewPupilsEligibleForLivePinGeneration' : 'vewPupilsEligibleForFamiliarisationPinGeneration'
-    const sql = `SELECT 
-                  * 
-                FROM ${sqlService.adminSchema}.${view}
-                WHERE school_id=@schoolId                 
+    let sql
+    if (!isLiveCheck) {
```

Comment:
Even though it's probably late at this stage, I still think data service layer should be logic free and encapsulate any logic related code (like if statements) in the caller service method that could potentially call different data service methods.

Reply:
noted for future work